hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/ConsensusValidatorImpl.java

```diff
@@ -0,0 +1,130 @@
+package com.hedera.mirror.importer.downloader;
+
+/*-
+ * ‌
+ * Hedera Mirror Node
+ * ​
+ * Copyright (C) 2019 - 2022 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Multimap;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.Collection;
+import java.util.HashMap;
+import javax.inject.Named;
+import lombok.CustomLog;
+import lombok.RequiredArgsConstructor;
+
+import com.hedera.mirror.common.domain.entity.EntityId;
+import com.hedera.mirror.importer.addressbook.AddressBookService;
+import com.hedera.mirror.importer.domain.FileStreamSignature;
+import com.hedera.mirror.importer.exception.SignatureVerificationException;
+import com.hedera.mirror.importer.repository.NodeStakeRepository;
+
+@CustomLog
+@Named
+@RequiredArgsConstructor
+public class ConsensusValidatorImpl implements ConsensusValidator {
+    private final AddressBookService addressBookService;
+    private final CommonDownloaderProperties commonDownloaderProperties;
+    private final NodeStakeRepository nodeStakeRepository;
+
+    /**
+     * Validates that the signature files satisfy the consensus requirement:
+     * <ol>
+     *  <li>If NodeStakes are within the NodeStakeRepository, at least 1/3 of the total node stake amount has been
+     *  signature verified.</li>
+     *  <li>If no NodeStakes are in the NodeStakeRepository, At least 1/3 signature files are present</li>
+     * </ol>
+     *
+     * @param signatures a list of signature files which have the same filename
+     * @throws SignatureVerificationException
+     */
+    @Override
+    public void validate(Collection<FileStreamSignature> signatures) throws SignatureVerificationException {
+        Multimap<String, FileStreamSignature> signatureHashMap = HashMultimap.create();
+        for (var signature : signatures) {
+            if (signature.getStatus() == FileStreamSignature.SignatureStatus.VERIFIED) {
+                signatureHashMap.put(signature.getFileHashAsHex(), signature);
+            }
+        }
+
+        var filename = signatures.stream().map(FileStreamSignature::getFilename).findFirst().orElse("unknown");
+        if (BigDecimal.ZERO.compareTo(commonDownloaderProperties.getConsensusRatio()) == 0
```

Comment:
nit: Believe it can be simplified to equals
```suggestion
        if (BigDecimal.ZERO.equals(commonDownloaderProperties.getConsensusRatio())
```

Reply:
Good idea, I've made this change.